src/printer.js

```diff
@@ -3998,7 +3998,13 @@ function printJSXChildren(path, options, print, jsxWhitespace) {
 
       const next = n.children[i + 1];
       const followedByJSXElement = next && !isLiteral(next);
-      if (followedByJSXElement) {
+      const followedByJSXWhitespace =
+        next &&
+        next.type === "JSXExpressionContainer" &&
+        next.expression.type === "Literal" &&
```

Comment:
might need to also support `StringLiteral`? I'm not up on the latest of the prettier codebase. Glancing up a line it looks like `isLiteral(next.expression)` might be what you want